### ISSUE TITLE
I need to use map function over an array of object and translate them together. 

### ISSUE THREAD

**Issue by salwacynthia** [i18next/react-i18next#1003]
I have en.json and de.json file which are succefully translating usin react-i18next. Now I have an array of object .... which I need to show in a list and translate them. I am trying the code below inside my class component, but it is getting error. I am not getting where is the mistake. 

const notificationSettingsContent = [
  {
  name: {
      id: "Id1r",
       defaultMessage: "hello1"
     },
     type: "Answer"
   },
  {
    name: {
      id: "idr2",
      defaultMessage: "hello2"
    },
    type: "answer"
  } 
]; 

And here is my component ::

class Container extends Component {
  static route = route;

  state = {
    account: {},
    notificationSettings: {}
  }

  componentDidMount() {
    this.fetch();
  }

  componentWillReceiveProps() {
    this.fetch();
  }

  fetch = () => getCurrentAccount().then((({ data }) => {
    this.setState({
      account: data,
      notificationSettings: (data || {}).notificationSettings
    });
  }));

  handleChange = (field, value) => {
    const notificationSettings = this.state.notificationSettings || {};

  updateAccount({ ...notificationSettings, [field]: value }).then(data => this.setState({
      notificationSettings: data
    }));
  };
handleClick (lang){
  this.props.i18n.changeLanguage(lang);
};

render() {
  const {
    classes,
    translatedObject = notificationSettingsContent.map(item =>{
      const ret = {
        ...item,
        name: item.name,
      };
    }),
    intl: {
        formatMessage
      }
    } = this.props;

    return (
      
      <div className={classes.root}>
        <AppBar position="static">
          <LocationToolBar />
        </AppBar>

        <ProfileHeader estate={this.props.currentEstate} />
        <nav>
          <button onClick={()=>this.handleClick('en')}>EN</button>
          <button onClick={()=>this.handleClick('de')}>DE</button>
        </nav>
        
       <Typography variant="h6" className={classes.content}>
                
          {/* {this.props.t('Welcome to React')} */}
          {this.props.t('profile.privacy.title.short')}
          {/* {formatMessage({ id: "profile.privacy.title.short" })} */}
        </Typography>
        
        <Typography className={classes.content}>
        {this.props.t('profile.privacy.title')}
         </Typography>
    
        <Typography variant="subtitle1" inline>
      
              {this.props.t('settings.private.issues.new')}
         </Typography>
        <Typography>
         {this.props.t.('notificationSettingsContent', {returnObject:true}).map(item=>{
            item = item.name;
          })}
     
        </Typography>
        
        <IntelSettings
          data={notificationSettingsContent}
          onChange={this.handleChange}
          defaultSettings={this.state.notificationSettings}
        />
      </div>
    );
  }
}

//with the EN and DE button data from json files are being translated, but I can do the same with the object inside the same .js file!

**Comment by jamuhl** [comment]
Sorry but very hard to follow those code snippets and see what does not work - can you provide a codesandbox?

Only thing I currently see you map the `notificationSettingsContent` rather strangely but that should not cause a problem I guess...

What error do you get? It might help to know the error to find the problem...

**Comment by salwacynthia** [comment]
Thanks for your reply. The error says, t is not a function at , where I am doing map.

**Comment by jamuhl** [comment]
```js
{this.props.t.('notificationSettingsContent', {returnObject:true}).map(item=>{
        item = item.name;
      })}
```

is that a typo? `this.props.t.(` should be `this.props.t(`

**Comment by salwacynthia** [comment]
sorry.. that was a typo.
without the typo I am getting this error message

this.props.t(...).map is not a function

**Comment by jamuhl** [comment]
check what is `this.props.t('notificationSettingsContent', {returnObject:true})` -> guess a string...not loaded yet, or wrong key?

**Comment by sairam6206** [comment]
this.props.t('notificationSettingsContent.0. nameofvarible);

you got value this after then you can try this

{this.props.t('notificationSettingsContent', {returnObject:true}).map(index,item)=>{
console.log(index.nameofvariable)
}}
